src/lib/components/controls/ControlsList.svelte

```diff
@@ -119,15 +119,83 @@
 				results = results.filter((control) => {
 					// Control must match all filters
 					return $activeFilters.every(filter => {
+						// Handle mapping-related fields specially
+						if (filter.fieldName === 'has_mappings') {
+							const hasMappings = control.mappings && control.mappings.length > 0;
+							const filterValue = getFilterValue(filter.value);
+							switch (filter.operator) {
+								case 'equals':
+									return hasMappings === (filterValue === 'true' || filterValue === true);
+								case 'not_equals':
+									return hasMappings !== (filterValue === 'true' || filterValue === true);
+								case 'exists':
+									return hasMappings;
+								case 'not_exists':
+									return !hasMappings;
+								default:
+									return true;
+							}
+						}
```

Comment:
In the UI the contains/not contains options still showed up for me. Maybe we can just simplify this to a true/false for has mappings (and remove any options besides "equals" basically?).